src/types/vendor/mongoose.d.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Additional type declarations for mongoose to fit our use case,
+ * to accommodate the non-standard but compatible use of types
+ * in schema registration
+ */
```

Comment:
could you add a few lines of explanation here on why this is necessary? again, would be very useful for future developers. hard to tell what's going on without context.

Reply:
Done this by annotating the class and function